### PR TITLE
Warn validators about stake deactivation (rebased)

### PR DESCRIPTION
**Pull request by cyberbono3**
|Fixes] #5773
|Part of] #5955 

This PR implements warning validators  about stake deactivation.

### CODE REVIEW

**Comment by cyberbono3** [comment]
@mvines  can you review it , please ?

File: core/src/validator.rs [hunk 1]
```diff
@@ -201,6 +206,33 @@ impl Validator {
...
+                    .and_then(|account| {
+                        account.iter().find(|(pubkey, (stake, account))| {
+                            **pubkey == id
+                                && *stake > 0
+                                && VoteState::deserialize(&account.data).is_ok()
+                        })
+                    })
+                    .map(|(_, (n, _))| *n);
+                match stake {
+                    Some(_val) => warn!("Validator with stake is exiting on first ^C"),
```
Comment (mvines) [inline core/src/validator.rs]: Let's change this warning to something more more like this:
```rust
warn!("Validator is staked!  Press ^C again to exit"),
```

File: core/src/validator.rs [hunk 2]
```diff
@@ -201,6 +206,33 @@ impl Validator {
 
         let blocktree = Arc::new(blocktree);
 
+        if config.enable_ctrl_c_handler {
+            let bank_forks = bank_forks.clone();
+            let counter = Arc::new(AtomicUsize::new(0));
+            set_handler(move || {
+                if counter.fetch_add(1, Ordering::Relaxed) > 0 {
+                    warn!("Validator aborting on second ^C");
```
Comment (mvines) [inline core/src/validator.rs]: I think we can remove this `warn!` entirely, it doesn't really add much

**Comment by mvines** [comment]
\o/ it's green!

**Comment by mvines** [comment]
@cyberbono3 - can you please address my two `warn!` comments, then I think this is ready to land

File: core/src/validator.rs (hunk 1 above)
Comment (cyberbono3) [reply core/src/validator.rs]: got it

File: core/src/validator.rs (hunk 2 above)
Comment (cyberbono3) [reply core/src/validator.rs]: i have deleted warn! here

**Comment by cyberbono3** [comment]
@mvines i have addressed your two comments about warn!. Please review my changes.

**Pull request closed without merging**